### PR TITLE
Fix compiling for x86_64-unknown-linux-gnu when it's not the default target

### PR DESCRIPTION
**Pull request by jyn514**
Fixes https://github.com/rust-lang/docs.rs/issues/1556 by only special casing proc-macros instead of any crate targeting the host.

r? @Nemo157

### CODE REVIEW

File: src/docbuilder/rustwide_builder.rs [hunk 1]
```diff
@@ -935,4 +944,66 @@ mod tests {
...
+                &format!("{}/{}/index.html", target, crate_path),
+            )?;
+
+            let web = env.frontend();
+            let target_url = format!(
+                "/{}/{}/{}/{}/index.html",
+                crate_, version, target, crate_path
+            );
+
+            assert!(target_docs_present);
```
Comment (jyn514) [inline src/docbuilder/rustwide_builder.rs:1003]: I double checked and this does fail without the fix in this commit.

```
[2021-11-24T17:58:10Z DEBUG docs_rs::docbuilder::rustwide_builder] adding sources into database
[2021-11-24T17:58:10Z DEBUG docs_rs::db::add_package] Adding package into database
[2021-11-24T17:58:10Z DEBUG docs_rs::db::add_package] Adding doc coverage into database
[2021-11-24T17:58:10Z DEBUG docs_rs::db::add_package] Adding build into database
[2021-11-24T17:58:10Z INFO  docs_rs::db::add_package] Updating crate data for xingapi
[2021-11-24T17:58:10Z DEBUG docs_rs::docbuilder::rustwide_builder] cleaning old storage folder rustdoc/xingapi/0.3.3/
[2021-11-24T17:58:10Z DEBUG docs_rs::docbuilder::rustwide_builder] cleaning old storage folder sources/xingapi/0.3.3/
[2021-11-24T17:58:11Z ERROR docs_rs::web::page::templates] Failed to load rustc resource suffix: missing rustc version
[2021-11-24T17:58:11Z INFO  docs_rs::web] Running docs.rs web server on http://127.0.0.1:39737
thread 'docbuilder::rustwide_builder::tests::test_cross_compile_non_host_default' panicked at 'assertion failed: target_docs_present', src/docbuilder/rustwide_builder.rs:1004:13

failures:
    docbuilder::rustwide_builder::tests::test_cross_compile_non_host_default
```

**Comment by Nemo157** [comment]
I think we probably want to override `Metadata::targets` to only return a single target for proc-macros too (since we don't actually obey the target setting). I found a random proc-macro without a target override and it only has docs for the default target: <https://docs.rs/datatest-derive/0.6.3/datatest_derive/>.

**Comment by Nemo157** [comment]
`add_essential_files` needs to be updated to look in the target dir

```patch
-let source = build.host_target_dir().join("doc");
+let source = build.host_target_dir().join("doc").join(HOST_TARGET);

File: src/docbuilder/rustwide_builder.rs [hunk 2]
```diff
@@ -350,7 +350,11 @@ impl RustwideBuilder {
                     if res.result.successful {
                         if let Some(name) = res.cargo_metadata.root().library_name() {
                             let host_target = build.host_target_dir();
-                            has_docs = host_target.join("doc").join(name).is_dir();
+                            has_docs = if metadata.proc_macro {
+                                host_target.join("doc").join(name).is_dir()
+                            } else {
+                                host_target.join("doc").join(HOST_TARGET).join(name).is_dir()
```
Comment (Nemo157) [inline src/docbuilder/rustwide_builder.rs:356]: I think this needs to be `default_target`, in case the library does not build for the host target.

**Comment by jyn514** [comment]
Yay, this is finally passing tests :) this should be ready for rereview.

**Review by Nemo157 — APPROVED** [review]
All LGTM now. Tested locally with a variety of proc-macro/non-proc-macro/default=host/default=non-host crates and all build variants appear to work correctly.

**Comment by konan8205** [comment]
@Nemo157 Could you trigger the rebuild from 0.3.0 to 0.3.3 version? I want to check if the problem is solved.

**Pull request merged**